### PR TITLE
Introduce rebootstrapSize for VTGR

### PR DESCRIPTION
**Pull request by y5w**
Signed-off-by: narcsfz <y.wu4515@gmail.com>

## Description

Today, VTGR will only rebootstrap the group when all nodes for the shard are healthy. However, in some corner case (e.g., AZ outage), we cannot guarantee all the nodes will be online and we need a way to allow VTGR rebootstrap the group with partial nodes.

This PR introduces `rebootstrapSize` to VTGR. By default, it is init to 0 which will be ignored and old behavior above is preserved. Once it's overwritten, VTGR can tolerate errors from some nodes and rebootstrap with partial members. For example, with `expectedBootstrapSize = 5, rebootstrapSize = 4`: VTGR will tolerate 1 (5-4) failed node and rebootstrap the group with 4 healthy node in a shard.

## Related Issue(s)

See description above

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

**Comment by y5w** [comment]
Gentle nudge on the PR  😄

File: go/vt/vtgr/controller/repair_test.go [hunk 1]
```diff
@@ -119,6 +119,11 @@ func TestRepairShardHasNoGroup(t *testing.T) {
 				{MemberHost: "", MemberPort: "NULL", MemberState: "OFFLINE", MemberRole: ""},
 			}, topodatapb.TabletType_REPLICA},
 		}},
+		{"raise error when there is not enough members", 0, "vtgr repair: unsafe to bootstrap group", []data{
```
Comment (deepthi) [inline go/vt/vtgr/controller/repair_test.go:122]:
```suggestion
		{"raise error when there are not enough members", 0, "vtgr repair: unsafe to bootstrap group", []data{
```

File: go/vt/vtgr/controller/repair_test.go [hunk 2]
```diff
@@ -338,6 +344,33 @@ func TestRepairShardHasInactiveGroup(t *testing.T) {
...
+				{MemberHost: "", MemberPort: "NULL", MemberState: "OFFLINE", MemberRole: ""},
+			}, true, getMysql56GTIDSet(sid1, "1-11"), topodatapb.TabletType_REPLICA},
+			{"", 0, "group", []db.TestGroupState{
+				{MemberHost: "", MemberPort: "NULL", MemberState: "OFFLINE", MemberRole: ""},
+			}, true, getMysql56GTIDSet(sid1, "1-10"), topodatapb.TabletType_REPLICA},
+			{testHost, testPort2, "group", []db.TestGroupState{
+				{MemberHost: "", MemberPort: "NULL", MemberState: "OFFLINE", MemberRole: ""},
+			}, true, getMysql56GTIDSet(sid1, "1-9"), topodatapb.TabletType_REPLICA},
+		}},
+		{"no error on two unreachable mysql with allowUnhealthyNodeOnReboot", "", testPort2, 1, []data{
```
Comment (deepthi) [inline go/vt/vtgr/controller/repair_test.go:358]: mysql -> mysqls?
or mysqlds?

File: go/vt/vtgr/vtgr.go [hunk 3]
```diff
@@ -191,6 +193,18 @@ func (vtgr *VTGR) GetCurrentShardStatuses() []controller.ShardStatus {
 	return result
 }
 
+// OverrideRebootstrapGroupSize force override the group size used in safety check for rebootstrap
```
Comment (deepthi) [inline go/vt/vtgr/vtgr.go:196]: force override -> forces an override

**Review by deepthi — COMMENTED** [review]
I had some nits on the comments.
I think the PR does also need a rebase on main / merge from main because a required CI check is missing.

**Review by deepthi — APPROVED** [review]

**Pull request merged**